src/dreyfus_index_updater.erl

```diff
@@ -31,6 +31,14 @@ update(IndexPid, Index) ->
     erlang:put(io_priority, {view_update, DbName, IndexName}),
     {ok, Db} = couch_db:open_int(DbName, []),
     try
+        {ok, IdxPurgeSeq} = clouseau_rpc:get_purge_seq(IndexPid),
+        FoldFun = fun(PurgeSeq, {Id, _Revs}, Acc) ->
+            clouseau_rpc:delete(IndexPid, Id),
```

Comment:
I don't think this is the correct behaviour. The purge API is on id/rev pairs, the request might leave some revisions alone, in which case the index should reflect that view. This code appears to remove the document from the index in all cases.